octobot_trading/personal_data/portfolios/assets/future_asset.py

```diff
@@ -80,14 +95,20 @@ def update(self, total=constants.ZERO, available=constants.ZERO, position_margin
         if available == constants.ZERO and position_margin == constants.ZERO \
                 and total == constants.ZERO and unrealized_pnl == constants.ZERO:
             return False
-        self.initial_margin += self._ensure_update_validity(self.initial_margin, initial_margin)
-        self.position_margin = self._ensure_not_negative(self.position_margin + position_margin)
-        self.order_margin = self._ensure_not_negative(self.order_margin + available)
-        self.unrealized_pnl += unrealized_pnl
-        self.wallet_balance += self._ensure_update_validity(self.wallet_balance, total)
-        self._update_available()
-        self._update_total()
-        return True
+
+        previous_asset = copy.copy(self)
+        try:
+            self.initial_margin += self._ensure_update_validity(self.initial_margin, initial_margin)
+            self.position_margin = self._ensure_not_negative(self.position_margin + position_margin)
+            self.order_margin = self._ensure_not_negative(self.order_margin + available)
+            self.unrealized_pnl += unrealized_pnl
+            self.wallet_balance += self._ensure_update_validity(self.wallet_balance, total)
+            self._update_available()
+            self._update_total()
+            return True
+        except errors.PortfolioNegativeValueError as negative_value_error:
```

Comment:
considering this try/except block is used 3 times, should we put it into a context manager ?